actions/websocket_actions.test.jsx

```diff
@@ -930,21 +969,199 @@ describe('handlePluginEnabled/handlePluginDisabled', () => {
-            const dispatchRemovedArg = store.dispatch.mock.calls[0][0];
-            expect(dispatchRemovedArg.type).toBe(ActionTypes.REMOVED_WEBAPP_PLUGIN);
+            expect(store.dispatch).toHaveBeenCalledTimes(3);
+            expect(console.error).toHaveBeenCalledTimes(0);
+        });
+    });
+
+    describe('loadPluginsIfNecessary', () => {
+        const baseManifest = {
+            name: 'Demo Plugin',
+            description: 'This plugin demonstrates the capabilities of a Mattermost plugin.',
+            version: '0.2.0',
+            min_server_version: '5.12.0',
+            server: {
+                executables: {
+                    'linux-amd64': 'server/dist/plugin-linux-amd64',
+                    'darwin-amd64': 'server/dist/plugin-darwin-amd64',
+                    'windows-amd64': 'server/dist/plugin-windows-amd64.exe',
+                },
+            },
+            webapp: {
+                bundle_path: 'webapp/dist/main.js',
+            },
+        };
+
+        let getWebappPlugins;
+        let loadPluginsIfNecessary;
+
+        beforeEach(async () => {
+            console.log = jest.fn();
+            console.error = jest.fn();
+
+            const {Client4} = require('mattermost-redux/client'); // eslint-disable-line global-require
+            getWebappPlugins = Client4.getWebappPlugins;
+            loadPluginsIfNecessary = jest.requireActual('plugins').loadPluginsIfNecessary;
+
+            document.createElement = jest.fn();
+            document.getElementsByTagName = jest.fn();
+            document.getElementsByTagName.mockReturnValue([{
+                appendChild: jest.fn(),
+            }]);
+        });
+
+        test('should remove unused old plugins when socket reconnects', async () => {
+            const manifest = {
+                ...baseManifest,
+                id: 'com.mattermost.demo-2-plugin',
+            };
+
+            const initialize = jest.fn();
+            window.plugins = {
+                [manifest.id]: {
+                    initialize,
+                },
+            };
+            const mockScript = {};
+            document.createElement.mockReturnValue(mockScript);
+
+            expect(mockScript.onload).toBeUndefined();
+
+            // Enable plugin
+            handlePluginEnabled({data: {manifest}});
+            mockState.plugins.manifests = {[manifest.id]: manifest};
+
+            const dispatchReceivedArg = store.dispatch.mock.calls[0][0];
+            expect(dispatchReceivedArg.type).toBe(ActionTypes.ADD_WEBAPP_PLUGIN);
+            expect(dispatchReceivedArg.data).toBe(manifest);
+
+            expect(document.createElement).toHaveBeenCalledWith('script');
+            expect(document.createElement).toHaveBeenCalledTimes(1);
+
+            // Disable plugin with reload
+            mockState.entities.general.config.PluginsEnabled = 'true';
+            getWebappPlugins.mockImplementation(() => []);
+            await loadPluginsIfNecessary();
```

Comment:
can we also assert if the `loadPlugin` wasn't invoked when this is called, by checking if there was another `ADD_WEBAPP_PLUGIN` was invoked.